src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -104,14 +133,11 @@ public ServiceEndpoint(Uri endpoint, TokenCredential credential, EndpointType en
             {
                 throw new ArgumentNullException(nameof(endpoint));
             }
-            else if (endpoint.Scheme != Uri.UriSchemeHttp && endpoint.Scheme != Uri.UriSchemeHttps)
-            {
-                throw new ArgumentException("Endpoint scheme must be 'http://' or 'https://'");
-            }
+            CheckScheme(endpoint);
             AccessKey = new AadAccessKey(endpoint, credential);
-            AudienceBaseUrl = BuildAudienceBaseUrl(endpoint);
-            Endpoint = BuildServerEndpoint(endpoint);
-            ClientEndpoint ??= Endpoint;
+            AudienceBaseUrl = BuildAudienceBaseUrlEndWithSlash(endpoint);
```

Comment:
For this AAD related ctor, does it also make sense to support Client/Server endpoint? Can these ctors share the same inner ctor?